assets/js/translator.js

```diff
@@ -447,12 +447,33 @@ function getPairs() /*: JQueryPromise<any> */ {
             }
         });
 
-        for(var srcLang in pairs) {
-            // Default for new users is first available pair
-            curSrcLang = srcLang;
-            curDstLang = pairs[srcLang][0];
-            break;
+        // Default for new users is first available Browser Preference Language pair
+        var brwsLangs = navigator.languages; // Chrome, Mozilla and Safari
+        var prefLang;
+        $.each(brwsLangs, function(index) {
+            if(languages[brwsLangs[index].substr(0, 2)] !== undefined) {
+                curSrcLang = iso639CodesInverse[brwsLangs[index].substr(0, 2)];
+                autoSelectDstLang();
+                if(curDstLang !== undefined) {
+                    prefLang = brwsLangs[index].substr(0, 2);
+                    return false;
+                }
+            }
+        });
+        if(typeof prefLang === 'undefined') {
+            if(!navigator.userlanguage && !navigator.browserlanguage) {
+                curSrcLang = locale;
+            }
+            else {
+                var bLang = navigator.userlanguage || navigator.browserlanguage; // Internet Explorer
+                curSrcLang = iso639CodesInverse[bLang];
```

Comment:
What happens when `bLang` isn't in the available source languages?

Reply:
oops, i forgot it. :(

Reply:
Done. :)